include/dali/core/dev_array.h

```diff
@@ -56,16 +56,20 @@ class DeviceArray {
   __host__ __device__ constexpr const T &operator[](ptrdiff_t index) const noexcept
   { return data_[index]; }
 
-  __host__ __device__ inline T *begin() noexcept { return data_; }
-  __host__ __device__ constexpr const T *begin() const noexcept { return data_; }
-  __host__ __device__ constexpr const T *cbegin() const noexcept { return data_; }
-  __host__ __device__ inline T *end() noexcept { return data_ + N; }
-  __host__ __device__ constexpr const T *end() const noexcept { return data_ + N; }
-  __host__ __device__ constexpr const T *cend() const noexcept { return data_ + N; }
-  __host__ __device__ constexpr size_t size() const noexcept { return N; }
-  __host__ __device__ constexpr bool empty() const noexcept { return N == 0; }
-  __host__ __device__ inline T *data() noexcept { return data_; }
-  __host__ __device__ constexpr const T *data() const noexcept { return data_; }
+  DALI_HOST_DEV inline iterator begin() noexcept { return data_; }
```

Comment:
If you change `__host__ __device__` to `DALI_HOST_DEV`, please do so for the entire class (or even file).

Reply:
Done